crypto/serializer/serializer_pkey.c

```diff
@@ -12,11 +12,14 @@
 #include <openssl/params.h>
 #include <openssl/serializer.h>
 #include <openssl/core_names.h>
+#include <openssl/safestack.h>
 #include "internal/provider.h"
 #include "internal/property.h"
 #include "crypto/evp.h"
 #include "serializer_local.h"
 
+DEFINE_STACK_OF_STRING()
```

Comment:
Make that `DEFINE_STACK_OF_CSTRING()`...

Reply:
Happy to change this.

Name duplication is necessary because changes could be made under lock between querying and using the names -- i.e. the name pointers could disappear in the meantime.

BTW: what's the difference between CSTRING and STRING?

Reply:
CSTRING is "const char" instead of "char", and is made exactly for the situation where you don't want to duplicate a name.

The name map supports addition only, not removal, and the namenum structure isn't moved around in memory.  That makes the name pointers a bit more trustable than you think.
The only time they are removed is when the namemap is freed.

Reply:
Isn't freeing possible while not under lock?

Reply:
It is...  and this comes down to how namemaps are used.  It is possible for some function to have its own, but the *normal* thing is to have it attached to the library context, and that will only be freed when the library context is destroyed.

Reply:
Okay, I'll update.

Reply:
I've updated this to use CSTRING on the assumption things won't change underneath us (which seems reasonable).

Reply:
Okay for review again.  I'll wait for CIs to pass of course.